src/functions/build-function.ts

```diff
@@ -0,0 +1,133 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as fs from 'fs-extra';
+import * as yaml from 'js-yaml';
+import { FolderPick, ImageAndBuild } from './function-type';
+import { CliExitData } from '../cli/cmdCli';
+import { knExecutor } from '../cli/execute';
+import { FuncAPI } from '../cli/func-api';
+import { ExistingWorkspaceFolderPick } from '../util/existing-workspace-folder-pick';
+import { getStderrString } from '../util/stderrstring';
+
+async function executeBuildCommand(location: string, image: string, builder?: string): Promise<void> {
+  const result: CliExitData = await knExecutor.execute(FuncAPI.buildFunc(location, image, builder));
+  if (result.error) {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    vscode.window.showErrorMessage(`Fail to build Project Error: ${getStderrString(result.error)}`);
+    return null;
+  }
+  // eslint-disable-next-line @typescript-eslint/no-floating-promises
+  vscode.window.showInformationMessage('Function successfully build.');
+}
+
+async function functionBuilder(image: string): Promise<ImageAndBuild> {
+  const builder = await vscode.window.showInputBox({
+    ignoreFocusOut: true,
+    prompt: 'Provide Buildpack builder, either an as a an image name or a mapping name.',
+  });
+  if (!builder) {
+    return null;
+  }
+  return { image, builder };
+}
+
+async function functionImage(selectedFolderPick: vscode.Uri): Promise<ImageAndBuild> {
+  const imageList = [`$(plus) Provide new image`];
+  let funcData;
+  try {
+    const funcYaml = await fs.readFile(path.join(selectedFolderPick.fsPath, 'func.yaml'), 'utf-8');
+    funcData = yaml.safeLoadAll(funcYaml);
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    if (funcData?.[0].image) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      imageList.push(funcData[0].image);
+    }
+  } catch (error) {
+    // ignore
+  }
+  const imagePick =
+    imageList.length === 1
+      ? imageList[0]
+      : await vscode.window.showQuickPick(imageList, {
+          canPickMany: false,
+          ignoreFocusOut: true,
+        });
+  if (!imagePick) {
+    return null;
+  }
+  if (imagePick === `$(plus) Provide new image`) {
```

Comment:
I do not think $(plus) works with parens, should be curly braces.

Reply:
<img width="818" alt="Screenshot 2021-07-21 at 10 13 30 PM" src="https://user-images.githubusercontent.com/9924513/126527002-4f87553e-72c4-4d2f-8876-7e2954c017c1.png">

It works and am using $(plus) for icon in showQuickPick. if the user clicks on `$(plus) Provide new image` then I show showInputBox to enter a new image.

Reply:
ah, sorry, I thought you were adding a variable in